hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -2058,6 +2058,15 @@ vlapic_x2apic_pt_icr_access(struct acrn_vcpu *vcpu, uint64_t val)
 	} else {
 		dmask = vlapic_calc_dest(vcpu, shorthand, (dest == 0xffffffffU), dest, phys, false);
 
+		/**
+		 * If "Destination Shorthand" field of ICR (x2apic) is set as 10B (All Including Self) or
+		 * 11B (All Excluding Self), the hypervisor emits IPI to each vCPU in the VM,
+		 * so "Destination Shorthand" field of the ICR value should be set as 00B (No Shorthand).
+		 * If this field is not set as 00B and delivery mode is not "INIT" or "Start Up", this IPI
+		 * will be sent to each physical CPU core in the platform according to 10.6.1 Interrupt
+		 * Command Register (ICR), Vol 3, SDM.
+		 */
```

Comment:
we shall ensure "Destination Shorthand" field of ICR be 00B, as the emulation is done through sending IPI to each VCPU in dmask one by one

Reply:
Get it, thanks! Have updated comments, please help double check it.